examples/python3/wt_session.py

```diff
@@ -0,0 +1,534 @@
+        if (api_response.status_code != 302) or (
+            api_response.headers["Location"] is None
+        ):
+            print("\tfailed - clientLogin POST did not return expected 302 Redirect.")
+
+            # On failure, the response content will be a redisplay of the web page with the
+            # API Client Login form.
+            # print "Response status_code = "+str(api_response.status_code)
+            # print ("Header: "+str(api_response.headers))
+            # print ("Content:"+api_response.text)
+
+            self._authenticated = False
+            return self._authenticated
+
+        matches = re.search("authcode=(.*)", api_response.headers["Location"])
+        if matches is None:
+            print("\tfailed - clientLogin POST did not return authcode")
+            print("\t\tResponse status_code = " + str(api_response.status_code))
+            print("\t\tHeader: " + str(api_response.headers))
+
+            self._authenticated = False
+            return self._authenticated
+
+        self._authcode = matches.group(1)
+
+        # Step 2 - POST back the authcode we got. This completes the login/session setup
+        # at api.wikitree.com. Since we use the same Session for the post, the cookies are all
+        # saved. A success here is a 200 and we'll have WikiTree session cookies.
+        post_data = {"action": "clientLogin", "authcode": self._authcode}
+        api_response = self._api_session.post(
+            API_URL,
+            data=post_data,
+            allow_redirects=False,
+            auth=("wikitree", "wikitree"),
+        )
+        if api_response.status_code != 200:
+            print("clientLogin(authcode) failed.")
+            print("Response status_code = " + str(api_response.status_code))
+            print("Header: " + str(api_response.headers))
+            self._authenticated = False
+            return self._authenticated
+
+        # The cookies set by the API look like:
+        # {
+        #   'wikitree_wtb__session': '<<session_id>>',
+        #   'wikitree_wtb_Token': '<<token>>',
+        #   'wikidb_wtb_UserName': '<<WikiTree ID>>',
+        #   'wikitree_wtb_UserID': '<<WikiTree user_id>>'
+        # }
+        self._wt_cookie = self._api_session.cookies.get_dict()
+        if self._wt_cookie is None:
+            print("clientLogin(authcode) failed -- No Cookies.")
+            self._authenticated = False
+            return self._authenticated
+
+        self._user_name = self._wt_cookie["wikidb_wtb_UserName"]
+        self._user_id = self._wt_cookie["wikidb_wtb_UserID"]
+
+        print(f"Authentication succeeded: {self._user_name} {self._user_id}")
+
+        self._authenticated = True
+        return self._authenticated
+
+    def _do_post(self, post_data) -> dict:
+        """
+        A convenience function to do the actual POST.
+        Returns empty dictionary if not authenticated.
+        """
+        if not self._authenticated:
+            return None
```

Comment:
```suggestion
            return {}
```

Reply:
Corrected and added JSON decoding exception handling.